homeassistant/components/radarr/sensor.py

```diff
@@ -95,154 +76,137 @@
-    entities = [
-        RadarrSensor(config, radarr, description)
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Radarr sensors based on a config entry."""
+    async_add_entities(
+        RadarrSensor(hass.data[DOMAIN][entry.entry_id], description)
         for description in SENSOR_TYPES
-        if description.key in config[CONF_MONITORED_CONDITIONS]
-    ]
-    async_add_entities(entities, True)
+        if description.key != "wanted"
+    )
 
 
-class RadarrSensor(SensorEntity):
+class RadarrSensor(RadarrEntity, SensorEntity):
     """Implementation of the Radarr sensor."""
 
+    coordinator: RadarrDataUpdateCoordinator
+
     def __init__(
         self,
-        conf: ConfigType,
-        radarr: RadarrClient,
+        coordinator: RadarrDataUpdateCoordinator,
         description: SensorEntityDescription,
     ) -> None:
         """Create Radarr entity."""
+        super().__init__(coordinator)
         self.entity_description = description
-        self.radarr = radarr
-        self.included = cast(list, conf.get(CONF_INCLUDED))
-        self.days = int(cast(str, conf.get(CONF_DAYS)))
-        self.data: Any = None
         self._attr_name = f"Radarr {description.name}"
-        if description.key == "diskspace":
-            self._attr_native_unit_of_measurement = conf.get(CONF_UNIT)
+        self._attr_unique_id = f"{coordinator.config_entry.entry_id}_{description.name}"
 
     @property
-    def extra_state_attributes(self) -> dict[str, StateType]:
+    def extra_state_attributes(self) -> dict[str, StateType | datetime]:
         """Return the state attributes of the sensor."""
-        attributes: dict[str, None | str | int | float] = {}
-        if self.entity_description.key == "upcoming":
-            return {to_key(movie): release_date(movie) for movie in self.data}
         if self.entity_description.key == "commands":
-            return {command.name: command.status for command in self.data}
+            return {cmd.name: cmd.status for cmd in self.coordinator.commands}
         if self.entity_description.key == "diskspace":
-            for data in self.data:
-                free_space = to_unit(data.freeSpace, self.native_unit_of_measurement)
-                total_space = to_unit(data.totalSpace, self.native_unit_of_measurement)
-                attributes[data.path] = "{:.2f}/{:.2f}{} ({:.2f}%)".format(
-                    free_space,
-                    total_space,
-                    self.native_unit_of_measurement,
-                    0 if total_space == 0 else free_space / total_space * 100,
-                )
-        elif self.entity_description.key == "movies":
-            return {to_key(movie): movie.hasFile for movie in self.data}
+            return self.get_strings(cast(str, self.native_unit_of_measurement))
+        if self.entity_description.key == "movies":
```

Comment:
Is this putting every movie in the attributes?

Reply:
Yes. I believe this was the previous behavior but likely to be a problem as you can imagine. What about a limit to say 1000? Should this be user configurable?

Reply:
That data does not belong in the state machine.